docs/standard/commandline/snippets/define-commands/csharp/Program2.cs

```diff
@@ -0,0 +1,191 @@
+    }
+
+    static async Task GlobalOption(string[] args)
+    {
+        // <defineglobal>
+        var delayOption = new Option<int>
+            ("--delay", "An option whose argument is parsed as an int.");
+        var messageOption = new Option<string>
+            ("--message", "An option whose argument is parsed as a string.");
+
+        var rootCommand = new RootCommand();
+        rootCommand.AddGlobalOption(delayOption);
+        rootCommand.Add(messageOption);
+
+        var subCommand1 = new Command("sub1", "First level subcommand");
+        rootCommand.Add(subCommand1);
+
+        var subCommand1a = new Command("sub1a", "Second level subcommand");
+        subCommand1.Add(subCommand1a);
+
+        subCommand1a.SetHandler((int delayOptionValue) =>
+        {
+            Console.WriteLine($"--delay = {delayOptionValue}");
+        },
+        delayOption);
+
+        await rootCommand.InvokeAsync(args);
+        // </defineglobal>
+
+        Console.WriteLine("Request help for second level subcommand.");
+        await rootCommand.InvokeAsync("sub1 sub1a -h");
+        Console.WriteLine("Global option in second level subcommand.");
+        await rootCommand.InvokeAsync("sub1 sub1a --delay 42");
+    }
+
+    static async Task RequiredOption(string[] args)
+    {
+        // <requiredoption>
+        var endpointOption = new Option<Uri>("--endpoint") { IsRequired = true };
+        var command = new RootCommand();
+        command.Add(endpointOption);
+
+        command.SetHandler(
+            (Uri? uri) =>
+            {
+                Console.WriteLine(uri?.GetType());
+                Console.WriteLine(uri?.ToString());
+            }, endpointOption);
+
+        await command.InvokeAsync(args);
+        // </requiredoption>
+        Console.WriteLine("Provide required option");
+        await command.InvokeAsync("--endpoint https://contoso.com");
+    }
+
+    static async Task HiddenOption(string[] args)
+    {
+        // <hiddenoption>
+        var endpointOption = new Option<Uri>("--endpoint") { IsHidden = true };
+        var command = new RootCommand();
+        command.Add(endpointOption);
+
+        command.SetHandler(
+            (Uri? uri) =>
+            {
+                Console.WriteLine(uri?.GetType());
+                Console.WriteLine(uri?.ToString());
+            }, endpointOption);
+
+        await command.InvokeAsync(args);
+        // </hiddenoption>
+        Console.WriteLine("Request help for hidden option.");
+        await command.InvokeAsync("-h");
+        Console.WriteLine("Provide hidden option.");
+        await command.InvokeAsync("--endpoint https://contoso.com");
+    }
+
+    static async Task FromAmong(string[] args)
+    {
+        // <staticlist>
+        var languageOption = new Option<string>(
+            "--language",
+            "An option that that must be one of the values of a static list.")
+                .FromAmong(
+                    "csharp",
```

Comment:
Can we include VB on principle, unless the context is one we know VB doesn't support (I'm jumping around a bit in my review)/